plugins/modules/digital_ocean_domain.py

```diff
@@ -162,6 +161,19 @@ def core(module):
             if 'message' in domain:
                 module.fail_json(changed=False, msg=domain['message'])
             else:
+                # We're at the mercy of a backend process which we have no visibility into:
+                # https://developers.digitalocean.com/documentation/v2/#create-a-new-domain
+                #
+                # In particular: "Keep in mind that, upon creation, the zone_file field will
+                # have a value of null until a zone file is generated and propagated through
+                # an automatic process on the DigitalOcean servers."
+                #
+                # Arguably, it's nice to see the records versus null, so, we'll just try a
+                # few times before giving up and returning null.
+                for i in range(5):
+                    if do_manager.domain_record()['domain']['zone_file'] is not None:
+                        module.exit_json(changed=True, domain=do_manager.domain_record()['domain'])
```

Comment:
Since it looks like the `domain_record()` function calls into the REST API, I think it might be better to use the object that was returned from the previous call. Maybe something more like:
```
record = do_manager.domain_record()
if 'domain' in record:
    domain = record['domain']
    if 'zone_file' in domain:
        module.exit_json(changed=True, domain=domain)

Reply:
`if 'zone_file' in domain`  will be always true according to official API docs, so this implementation will not work as expected.

Reply:
Oh, one more thing. Due to the lack of error checking from the `domain_record()` function, you should probably also and a checks for None:
```
record = do_manager.domain_record()\
if record is not None and 'domain' in record:
    domain = record['domain']
    if domain is not None and 'zone_file' in domain:
        module.exit_json(changed=True, domain=domain)
```

Reply:
I agree with both of you on this -- I'll add the existential checking for now. I like @tadeboro suggestion of revisiting this down the road and making the API calls more robust.

Reply:
This really comes down to the question of "how do we know whether or not the async process has updated the record"? In the case of a "new domain", it's not that difficult: it's either present, or not. (I didn't check how that will look in the returned JSON; I was only worried about Python exploding when trying to dereference `None`.) In the case of an update, I suppose you'd have to check whether your updates are present or not.

Reply:
Checking for an update might prove itself to be unfeasible here. It would probably mean parsing the contents of the zone_file context, which sounds like a big no ;)